src/components/Game.tsx

```diff
@@ -1,44 +1,26 @@
-import * as React from "react";
-import { play, GESTURES, Gesture, Result } from "../game/game";
+import React, { useEffect, useState } from "react";
+import { determineOutcome, Gesture, Outcome } from "../game";
+import { GameProvider } from "../context";
 
-type State = {
-  winner: Result;
-  hasPlayed: boolean;
-};
-
-export default class Game extends React.Component<{}, State> {
-  state = {
-    winner: "Draw" as Result,
-    hasPlayed: false
-  };
+export const Game: React.FunctionComponent = ({ children }) => {
+  const [outcome, setOutcome] = useState(Outcome.None);
+  const [playerGesture, setPlayerGesture] = useState<Gesture | null>(null);
 
-  play(playerGesture: Gesture) {
-    const winner = play(playerGesture);
+  useEffect(() => {
+    if (playerGesture !== null) {
+      setOutcome(determineOutcome(playerGesture));
+    }
+  }, [playerGesture])
 
-    this.setState({ winner, hasPlayed: true });
-  }
+  const context = {
+    outcome,
+    playerGesture,
+    setPlayerGesture,
```

Comment:
why does the `GameProvider` need the `setPlayerGesture` function? I think I've not followed the logic correctly here?